components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -168,66 +170,51 @@ public Response issueAccessToken(@Context HttpServletRequest request,
 
             try {
-                CarbonOAuthTokenRequest oauthRequest = new CarbonOAuthTokenRequest(httpRequest);
-                // exchange the access token for the authorization grant.
-                OAuth2AccessTokenRespDTO oauth2AccessTokenResp = getAccessToken(oauthRequest);
-                // if there BE has returned an error
-                if (oauth2AccessTokenResp.getErrorMsg() != null) {
-                    // if there is an auth failure, HTTP 401 Status Code should be sent back to the client.
-                    if (OAuth2ErrorCodes.INVALID_CLIENT.equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleBasicAuthFailure();
-                    } else if ("sql_error".equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleSQLError();
-                    } else if (OAuth2ErrorCodes.SERVER_ERROR.equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleServerError();
-                    } else {
-                        // Otherwise send back HTTP 400 Status Code
-                        OAuthResponse.OAuthErrorResponseBuilder oAuthErrorResponseBuilder = OAuthASResponse
-                                .errorResponse(HttpServletResponse.SC_BAD_REQUEST)
-                                .setError(oauth2AccessTokenResp.getErrorCode())
-                                .setErrorDescription(oauth2AccessTokenResp.getErrorMsg());
-                        OAuthResponse response = oAuthErrorResponseBuilder.buildJSONMessage();
-
-                        ResponseHeader[] headers = oauth2AccessTokenResp.getResponseHeaders();
-                        ResponseBuilder respBuilder = Response
-                                .status(response.getResponseStatus());
-
-                        if (headers != null && headers.length > 0) {
-                            for (int i = 0; i < headers.length; i++) {
-                                if (headers[i] != null) {
-                                    respBuilder.header(headers[i].getKey(), headers[i].getValue());
-                                }
-                            }
-                        }
-
-                        return respBuilder.entity(response.getBody()).build();
+                oauthRequest = new CarbonOAuthTokenRequest(httpRequest);
+            } catch (OAuthProblemException e) {
+                /*Since oltu library sends OAthProblemException upon real exception and input errors we need to show input errors when debugging
+                  and need to show the error logs when real exception thrown
+                  */
+                if (OAuthError.TokenResponse.INVALID_REQUEST.equalsIgnoreCase(e.getError()) || OAuthError.TokenResponse.UNSUPPORTED_GRANT_TYPE.equalsIgnoreCase(e.getError())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Invalid request or unsupported grant type:" + e.getError() + ", description: " + e.getDescription());
```

Comment:
grant type:" 
description:  Better if both are in the same format (add a spece after : )